### ISSUE TITLE
UI: Utilize FreeCAD icon as a super key to hide top menu and remove window top bar

### ISSUE THREAD

**Issue by AdrianCM95** [FreeCAD/FreeCAD#18630]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

This is an improvement request over the current user interface:

- to remove the window top bar to have more working space.
- to use the Freecad icon as a super key to hide the normal text menus under it (and also display a search bar to quickly find commands like https://github.com/FreeCAD/FreeCAD/issues/8290)

In order to call this function, a shortcut like  Superkey+F, Shift+F, (F,F)..., or whatever combination could be used.
I think this can help to speed up the workflow even without knowing all the shortcuts for the rest of the functions.

This is the current layout:
![image](https://github.com/user-attachments/assets/33f3ec2e-3811-4b81-8ac7-c502ff2dff33)

This is the proposed layout:
![image](https://github.com/user-attachments/assets/0aefd0b0-7b35-4c70-b10b-78357bf703f1)

### Full version info

```shell
OS: TUXEDO OS 3 (KDE/plasma/xcb)
Architecture: x86_64
Version: 1.0.0.39109 (Git) Conda AppImage
Build type: Release
Branch: (HEAD detached at 1.0.0)
Hash: 2fcc5317fe3aee96ca73475986a577719fc78e20
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: Spanish/Spain (es_ES)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Qt default
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tarman3** [comment]
https://github.com/Grubuntu/PieMenu

**Comment by maxwxyz** [comment]
Duplicate of https://github.com/FreeCAD/FreeCAD/issues/8290

**Comment by luzpaz** [comment]
Is it a duplicate though ? #8290 doesn't propose altering the top of the UI to save space, @maxwxyz

**Comment by AdrianCM95** [comment]
I would say it is not a full duplicate. The search bar is a duplicate, true. However, the button to hide the menu and removing the window top bar is different.

**Comment by maxwxyz** [comment]
Could you rephrase the original issue and title please

**Comment by AdrianCM95** [comment]
@maxwxyz I rephrased the contract a little bit, is it better now?

**Comment by yorikvanhavre** [comment]
This is more a duplicate of https://github.com/FreeCAD/FreeCAD/issues/12742

**Comment by luzpaz** [comment]
Right, can we merge this ticket in to 12742 ?

**Comment by yorikvanhavre** [comment]
Closing this one then, in favour of #12742

**Comment by luzpaz** [comment]
@AdrianCM95 would you mind posting your content from this ticket into #12742 ?
